### PR TITLE
Dashboards 2.0 base

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

The [new dashboard design](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=5958%3A45941) is mostly there, so this is the base for its implementation, implementing a new `InsightCard` component. Its usage is gated behind feature flag `dashboard-redesign`, so this should be safe for merging regardless of `InsightCard`'s current state.

TODOs for follow-up PRs:
- [x] fix and optimize dashboard sizing
- [x] analytics instrumentation (`data-attr`s and `eventUsageLogic`)
- [x] item resizing
- [x] all insight visualizations other than `LineGraph`
- [x] appropriate time range summary default values
- [x] expandable query summary
- [ ] skeleton variant for dashboard loading state
- [ ] refactor `updateItemColor` and `removeItem` to account for potential request errors
- [ ] use `LemonTag` instead of Ant `Tag`

## How did you test this code?

There's a handy Storybook story (though be wary of the somewhat buggy graphs – some odd Kea issue that does not occur in the app, didn't figure it out yet).
Screenshot 2021-12-15 at 12 25 50

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]
First thing: toggling the sidebar doesn't change the dashboard's width:


Then, I'm sure you all will give me feedback that I should have commented on Figma before the implementation, and now it's too late, but... 

1) I'm still not a fan of the grey background for any chart. It makes them look muddy/washed/less readable. Plus it makes it impossible to screenshot and then directly use them somewhere (e.g. in a white word doc), without resorting to the "paint bucket with white" tool in photoshop on every screenshot, which reduces quality and takes time.

2) I think these labels could look better on top of the charts. The page is not scannable right now, and I need to spend considerable effort to figure out what label goes with which chart. Unless you're spending effort to really see the borders between dashboard items, it's just super natural to connect the top label to the bottom chart, because this is how I read every other document/page everywhere: title -> content.

image

Basically, I noticed that many times in the small time I've had with the new design, that I read the title, look at the graph, and only then realise that there's another title that's the right one for the graph and I was reading something else before.

3) I'm sad to see the colors go away, as they added a bit of life to otherwise dull dashboards. A 4px left border doesn't do this justice, and now we're left with a lot of dull grey everywhere :/. 

Before:
image

After:
image

If I'm in the minority with these opinions, I'll concede, though I'd ask for a lot more opinions before proceeding. 

The code looks good though, so LGTM to push out behind a flag in order to get those opinions :).

**Comment by Twixes** [comment]
Using this, I kind of +1 that titles were more intuitive at the top, and the viz background being white would be more practical.

Regarding colors, I think the indication should be a bit more prominent, _but_ I like that the cards aren't colored completely. This way the dashboard looks more cohesive, and there are no edge cases with graph colors to be handled, so we'll be able to better curate colors too.

File: frontend/src/lib/components/InsightCard/InsightCard.scss [hunk 1]
```diff
@@ -0,0 +1,107 @@
+.InsightCard {
+    transition: border 200ms ease;
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+    height: 30rem;
+    border-radius: var(--radius);
```
Comment (paolodamico) [inline frontend/src/lib/components/InsightCard/InsightCard.scss:8]: Nit: Prefer SCSS variables over CSS global vars

**Review by paolodamico — APPROVED** [review]
Love this!

File: frontend/src/lib/components/InsightCard/InsightCard.scss (hunk 1 above)
Comment (Twixes) [reply frontend/src/lib/components/InsightCard/InsightCard.scss:8]: I'm of a quite strong opinion that CSS vars should be preferred unless the value really must be baked in at compile time (e.g. due to some calculations using Sass functions). CSS vars have been introduced ~5 years ago to replace all the various preprocessor-based solutions like Sass, and now have great browser support. The advantages are:
- no dependency on a preprocessor
- inheritance working just like with regular inherited CSS properties
- there's ALWAYS a single source of truth, even when working with a fully built frontend – no need to switch back and forth between the browser and editor, just edit the relevant `--variable` in devtools

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 2]
```diff
@@ -76,7 +76,8 @@ export const dashboardLogic = kea<dashboardLogicType<DashboardLogicProps>>({
         updateLayouts: (layouts: Layouts) => ({ layouts }),
         updateContainerWidth: (containerWidth: number, columns: number) => ({ containerWidth, columns }),
         saveLayouts: true,
-        updateItemColor: (insightId: number, color: string) => ({ insightId, color }),
+        updateItemColor: (insightId: number, color: string | null) => ({ insightId, color }),
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:79]: I did that initially, but this is still used by the old component where I don't want to mess with types much. Will definitely update later on.

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 3]
```diff
@@ -532,6 +539,11 @@ export const dashboardLogic = kea<dashboardLogicType<DashboardLogicProps>>({
         updateItemColor: async ({ insightId, color }) => {
             return api.update(`api/projects/${values.currentTeamId}/insights/${insightId}`, { color })
         },
+        removeItem: async ({ insightId }) => {
+            return api.update(`api/projects/${values.currentTeamId}/insights/${insightId}`, {
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:543]: That's a good point, I'll refactor this along with `removeItem` in a follow-up to guard against that.

**Comment by Twixes** [comment]
Fully agreed that it should resize with the sidebar @mariusandra. Though that has been the case with the current layout from the beginning, unrelated to the new `InsightCard`, so I'll address this separately to keep the diff of this sane.

File: frontend/src/lib/components/InsightCard/InsightCard.scss (hunk 1 above)
Comment (paolodamico) [reply frontend/src/lib/components/InsightCard/InsightCard.scss:8]: Alright those seem like pretty solid arguments, let's use CSS then
